x-pack/plugins/security_solution/cypress/tasks/api_calls/timelines.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Timeline } from '../../objects/timeline';
+
+export const createTimeline = async (timeline: Timeline): Promise<string[]> => {
+  const response = await cy
+    .request({
+      method: 'POST',
+      url: 'api/timeline',
+      body: {
+        timeline: {
+          columns: [
+            {
+              id: '@timestamp',
+            },
+            {
+              id: 'user.name',
+            },
+            {
+              id: 'event.category',
+            },
+            {
+              id: 'event.action',
+            },
+            {
+              id: 'host.name',
+            },
+          ],
+          kqlMode: 'filter',
+          kqlQuery: {
+            filterQuery: {
+              kuery: {
+                expression: timeline.query,
+                kind: 'kuery',
+              },
+            },
+          },
+          dateRange: {
+            end: '1577881376000',
+            start: '1514809376000',
+          },
+          description: timeline.description,
+          title: timeline.title,
+        },
+      },
+      headers: { 'kbn-xsrf': 'cypress-creds' },
+    })
+    .promisify();
+
+  const timelineId = response.body.data.persistTimeline.timeline.savedObjectId;
+  const timelineBody = response.body.data.persistTimeline.timeline;
+
+  return [timelineId, timelineBody];
```

Comment:
As per my comment about promises/errors, ideally this method just returns `cy.request()` and these custom types go away. However, in the case that they don't, I don't believe that `timelineBody` is a `string` here as the types indicate.

Reply:
We also don't seem to be _using_ `timelineBody`, so 🤷‍♂️